Document/0x06b-Basic-Security-Testing.md

```diff
@@ -343,60 +350,88 @@ $ ssh -p 2222 root@localhost
-Application: /private/var/mobile/Containers/Bundle/Application/A8BD91A9-3C81-4674-A790-AF8CDCA8A2F1/UnCrackable Level 1.app
-Data: /private/var/mobile/Containers/Data/Application/A8AE15EE-DC8B-4F1C-91A5-1FED35258D87
+iPhone:~ root# tar czvf /tmp/data.tgz /private/var/mobile/Containers/Data/Application/CC24A101-A668-4F77-B410-2FF47A281D05
+iPhone:~ root# exit
+$ scp -P 2222 root@localhost:/tmp/data.tgz .
 ```
 
-You can now simply archive the Data directory and pull it from the device with `scp`:
+##### Passionfruit
 
-```shell
-iPhone:~ root# tar czvf /tmp/data.tgz /private/var/mobile/Containers/Data/Application/A8AE15EE-DC8B-4F1C-91A5-1FED35258D87
-iPhone:~ root# exit
-$ scp -P 2222 root@localhost:/tmp/data.tgz .
+After starting Passionfruit you can select the app that is in scope for testing. There are various functions available, of which one is called "Files". When selecting it, you will get a listing of the directories of the app sandbox.
+
+When navigating through the directories and selecting a file, a TextViewer pop-up will show up that illustrates the data either as hex or text. When closing this pop-up you have various options available for the file, including:
+
+- Text viewer
+- SQLite viewer
+- Image viewer
+- Plist viewer
+- Download
+
+<img src="Images/Chapters/0x06b/Passionfruit_files.png" alt="Passiofruit File Options">
+
+##### Objection
+
+When you are starting objection you will find the prompt within the bundle directory.
+
+```bash
+org.owasp.MSTG on (iPhone: 10.3.3) [usb] # pwd print
+Current directory: /var/containers/Bundle/Application/DABF849D-493E-464C-B66B-B8B6C53A4E76/org.owasp.MSTG.app
+```
+
+Use the `env` command to get the directories of the app and navigate to the Documents directory. 
+
+```bash
+org.owasp.MSTG on (iPhone: 10.3.3) [usb] # cd /var/mobile/Containers/Data/Application/72C7AAFB-1D75-4FBA-9D83-D8B4A2D44133/Documents
+/var/mobile/Containers/Data/Application/72C7AAFB-1D75-4FBA-9D83-D8B4A2D44133/Documents
 ```
 
---ToDo: <https://github.com/OWASP/owasp-mstg/issues/1245>
+With the command `file download <filename>` you can download a file from the iOS device to your workstation and can analyse it afterwards.
+
+```bash
+org.owasp.MSTG on (iPhone: 10.3.3) [usb] # file download .com.apple.mobile_container_manager.metadata.plist
+Downloading /var/mobile/Containers/Data/Application/72C7AAFB-1D75-4FBA-9D83-D8B4A2D44133/.com.apple.mobile_container_manager.metadata.plist to .com.apple.mobile_container_manager.metadata.plist
+Streaming file from device...
+Writing bytes to destination...
+Successfully downloaded /var/mobile/Containers/Data/Application/72C7AAFB-1D75-4FBA-9D83-D8B4A2D44133/.com.apple.mobile_container_manager.metadata.plist to .com.apple.mobile_container_manager.metadata.plist
+```
+
+You can also upload files with `file upload <local_file_path>` to the iOS device, but this implementation is not fully stable at the moment and might produce an error. If that's the case file an issue.
```

Comment:
maybe say file an issue on objection's github? so that they don't file the objection issues to the MSTG :)